### DISCUSSION TITLE
OBS VirtualCam as virtual USB device

### DISCUSSION THREAD

**Discussion by tpeterke** [obsproject/obs-studio#5909] [General]
Hi OBS Developers,

Without any knowledge of the technology under the hood, please forgive if my question is stupid.
In my setup I run a Win10 guest on a Win10 host in VirtualBox.
I know I can install OBS in guest and it works too, but the guest is not powerful enough to support all the magic OBS can do for me.
When I setup OBS on the host, it runs smooth and perfect, but VirtualBox guest cannot see OBS's virtual cam because it's not an USB device (real or virtual).
I was reading about **"named pipes"(?)** and I'm not sure it helps at all, but named pipes can be configured as virtual ports in VirtualBox.

The only thing I know that it is possible somehow because XSplit does it somehow. Its virtual camera is visible in VirtualBox guest.
Can you please advise if/how to solve this?
Thank you.

-Peter

**Comment by Fenrirthviti** [comment]
We do not provide support over GitHub discussion, as explained in the stickied post here: https://github.com/obsproject/obs-studio/discussions/5206

Additionally, we do not support running OBS in a virtual environment. You can try asking in the Discord, but it's unlikely anyone will be able to assist, unfortunately.
